layer1/executor/task_executor.go

```diff
@@ -0,0 +1,354 @@
+package executor
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/alicenet/alicenet/consensus/db"
+	"github.com/alicenet/alicenet/constants"
+	"github.com/alicenet/alicenet/constants/dbprefix"
+	"github.com/alicenet/alicenet/layer1"
+	"github.com/alicenet/alicenet/layer1/executor/tasks"
+	"github.com/alicenet/alicenet/layer1/transaction"
+	"github.com/alicenet/alicenet/logging"
+	"github.com/alicenet/alicenet/utils"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/sirupsen/logrus"
+)
+
+type TaskExecutor struct {
+	sync.RWMutex
+	TxsBackup map[string]*types.Transaction `json:"transactions_backup"`
+	txWatcher transaction.Watcher           `json:"-"`
+	database  *db.Database                  `json:"-"`
+	logger    *logrus.Entry                 `json:"-"`
+}
+
+// newTaskExecutor creates a new TaskExecutor instance and recover the previous state from DB.
+func newTaskExecutor(txWatcher transaction.Watcher, database *db.Database, logger *logrus.Entry) (*TaskExecutor, error) {
+	taskExecutor := &TaskExecutor{
+		TxsBackup: make(map[string]*types.Transaction),
+		txWatcher: txWatcher,
+		database:  database,
+		logger:    logger,
+	}
+
+	err := taskExecutor.loadState()
+	if err != nil {
+		taskExecutor.logger.Warnf("could not find previous State: %v", err)
+		if err != badger.ErrKeyNotFound {
+			return nil, err
```

Comment:
Cover exit case?
This should unwind the main.

Reply:
This returns the error to the TaskManager and it is returning the error to the TaskHandler. Then the cf is called on the Handler (L:36)